### ISSUE TITLE
_table_sha1 depends on columns only available in MySQL 5.7

### ISSUE THREAD

**Issue by hepabolu** [hepabolu/mytap#15]
I need these tap tests to work on earlier versions as well, i.e. 5.5.x and 5.6.x. The function _table_sha1 concatenates info in multiple columns, a.o. 'generation_expression'. This column doesn't exist in version 5.5. and 5.6.

The code should accommodate for various versions.

**Comment by theory** [comment]
Quick aside: I've unwatched this repo. YOU GOT THIS!

**Comment by hepabolu** [comment]
Thanks @theory

**Comment by animalcarpet** [comment]
Helma, that function is probably worth having a chat about but doesn't strictly need to be in the release for the purpose of unit testing, so can be dropped. The purpose is to test if two schemas are identical by getting a SHA1 of the concatenated object definitions, obviously, it's version dependent. I'll remove it.

**Comment by hepabolu** [comment]
I think it's very useful and love to use it, so dropping would be a last resort.

**Comment by animalcarpet** [comment]
Not exactly. As a feature I don't think it sits comfortably with the unit test functions of mytap where we compare a schema object against a valid value that you can find somewhere in the information schema tables. The SHA1 you derive isn't a value you'll find anywhere in the information_schema - it is the information_schema.

I've incorporated this function as a part of a much larger one that derives a SHA1 from all objects in a schema, not just a single table, so had intended to remove it. I haven't committed the other function because of bugs in the information_schema itself and because I hadn't decided the appropriate place for it, either as part of mytap or a separate repo that depends on mytap. It'll obviously need a separate version for each MySQL release. I'll email you a copy if you want to try it out but it'll only work against 5.7 as it stands.

**Comment by animalcarpet** [comment]
On the duplication issue,  I'll try to give that some thought but I don't think it's going to mean too much duplication, more likely hiving some functions off into separate version dependent files and writing a different installer. It makes a lot of sense to support as many versions as possible going back to 5.5 but also the bleeding edge which I'm sure some people would already be contemplating for use (if only they had something to test it with).

**Comment by animalcarpet** [comment]
Leaving this in for now.
